packages/backend-defaults/src/entrypoints/auth/external/ExternalTokenHandler.ts

```diff
@@ -16,19 +16,63 @@
 
 import {
   BackstagePrincipalAccessRestrictions,
+  createServiceRef,
   LoggerService,
   RootConfigService,
 } from '@backstage/backend-plugin-api';
 import { NotAllowedError } from '@backstage/errors';
-import { LegacyTokenHandler } from './legacy';
+import { LegacyConfigWrapper, LegacyTokenHandler } from './legacy';
 import { StaticTokenHandler } from './static';
 import { JWKSHandler } from './jwks';
 import { TokenHandler } from './types';
+import { Config } from '@backstage/config';
+import { groupBy } from 'lodash';
+import { TokenTypeHandler } from './types';
 
 const NEW_CONFIG_KEY = 'backend.auth.externalAccess';
 const OLD_CONFIG_KEY = 'backend.auth.keys';
 let loggedDeprecationWarning = false;
 
+type LegacyTokenTypeHandler = {
+  type: 'legacy';
+  /**
+   * A factory function that takes all token configuration for a given type
+   * and returns a TokenHandler or an array of TokenHandlers.
+   */
+  factory: (
+    configs: (
+      | import('@backstage/config').Config
+      | { legacy: true; config: import('@backstage/config').Config }
+    )[],
+  ) => TokenHandler | TokenHandler[];
+};
+
+/**
+ * @public
+ * This service is used to decorate the default plugin token handler with custom logic.
+ */
+export const externalTokenTypeHandlersRef = createServiceRef<TokenTypeHandler>({
+  id: 'core.auth.externalTokenHandlers',
+  multiton: true,
+  // defaultFactory // :pepe-think: seems like is not possible to use defaultFactory with multiton
```

Comment:
🤔  seems like is not possible to use defaultFactory with multiton, should we look on this? maybe in. a separated PR anyways